### DISCUSSION TITLE
Option to disable CPack for cross-compiling ?

### DISCUSSION THREAD

**Discussion by ddscentral** [SoftEtherVPN/SoftEtherVPN#2062] [Q&A]
Hello,

In the past I did manage to cross-compile SoftEther VPN client (version 4.x) for 32-bit ARM architecture by manually tweaking the Makefile. It ran great on a 4G modem where it served as a transparent WiFi bridge to my home network. About twice as fast as OpenVPN.

Since you now use CMake in version 5.x it should in theory be easier to cross-compile SoftEther for a different architecture, but I do not see a good way to disable CPack without resorting to editing the build files. An option to disable packaging and just do a simple binary installation to a set location would be great.
 
If there is one already, do let me know.

**Comment by chipitsine** [comment]
we are open to contribution, if you consider making a PR.

as for disabling CPack, I do not have certain opinion. maybe it would be good to have some variable which simply disable it.
or maybe some condition, feel free to propose.

**Reply by ddscentral** [reply]
Thanks.
Will try cross-building with CPack disabled by commenting out, see how it goes.
I believe a simple variable to disable packaging should be enough, but I still need to take a closer look at how the new build system works.

Often, cross-compiling is done with an embedded system as a target and most of them do not use package managers at all. In my case, the target system uses buildroot (my old ARM target also used buildroot).

**Reply by chipitsine** [reply]
google suggests me the following  https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html
but as usual there may be caveats
